### PR TITLE
Hide screen reader value

### PR DESCRIPTION
**Pull request by efischer19**
This is a fix for [TNL-4009](https://openedx.atlassian.net/browse/TNL-4009)

~~The text was appearing oddly here, and it looks as if a very similar value close by in the template is being used as screen reader only (`.submission__answer__part__prompt__title` at line 482 in `_oa-base.scss`, versus `.submission__answer__part__text__title` which I just added at line 503). See https://github.com/edx/edx-ora2/blob/master/openassessment/templates/openassessmentblock/oa_submission_answer.html for the template that defines these classes. No other style references this `..._text_title` class, so I think it just got overlooked. The cleanest fix seems to be to apply the `%text-sr` style here as well.~~

~~The `oa_submission_answer.html` template is used elsewhere, so this change will hide a bit more text than just in the leaderboard. That text looked fairly unstyled though, so I still think it was always intended to be hidden as screen-reader text. I'll add some screenshots shortly.~~

### CODE REVIEW

**Comment by edx-webhook** [comment]
## PR Cover Letter
- [ ] A detailed description of the changes in the body of the PR
- [ ] If available, a link to the relevant JIRA ticket in the body of the PR
  - `[PROJ-XXXX](https://openedx.atlassian.net/browse/PROJ-XXXX)`

## Sandbox
- [ ] Build a sandbox for your branch and add a link here

## Testing
- [ ] i18n
- [ ] RTL
- [ ] Unit, integration, acceptance tests as appropriate
- [ ] Analytics
- [ ] Performance
- [ ] Database migrations are backwards-compatible

## Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [ ] Code review #1
- [ ] Code review #2
- [ ] Docs review (required for UI strings/error messages)
- [ ] UX review
- [ ] Accessibility review
- [ ] Product review

## Documentation considerations
- [ ] Are there user visible changes? If so, list who is affected, and include a pointer to the relevant code.
- [ ] Should this be mentioned in the release notes? If so, make sure that you included a description.

## DevOps assistance
If it's needed, or even if you suspect it might be needed, do any/all these things as early as possible. Don't surprise devops right before merging! For example, touching envs/aws.py or or envs/common.py are dead giveaways that you'll need assistance when pushing your changes out. Things that require devops attention include, but are not limited to:
- configuration changes
- feature flags
- migrations
Ways to get devops attention, pick-and-choose whichever is most appropriate for your specific PR:
- [ ] Tag devops on your PR (`@edx/devops`), either for review (more specific to devops = better) or just a general FYI
- [ ] File a support ticket

FYI: Tag anyone who might be interested in this PR here.

## Post-review
- [ ] Squash commits

**Comment by dianakhuang** [comment]
:+1:

**Comment by efischer19** [comment]
Here are some screenshots of the "not just in the leaderboard" text being changed.

Before, note how "Your response to the question above" seems unstyled:
screen shot 2016-02-19 at 2 42 53 pm

After, no text. Note that "Your Response" is still present (and styled!) above the prompt:
screen shot 2016-02-19 at 2 45 11 pm

**Comment by robrap** [comment]
@efischer19 I'll need to review the original ticket, but your images make me question this change.  It doesn't seem to be an improvement to me.

**Comment by cahrens** [comment]
I agree with Robert. The text adds clarity for all users, so we should style it correctly and always show it (in my opinion).

**Comment by efischer19** [comment]
Tagging @chris-mike, this is the PR being discussed.

File: openassessment/templates/openassessmentblock/oa_submission_answer.html [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
             <article class="submission__answer__part__prompt">
-                <h2 class="submission__answer__part__prompt__title">{% trans "The question for this section." %}</h2>
+                <h3 class="submission__answer__part__prompt__title">{% trans "The question for this section." %}</h3>
                 <div class="submission__answer__part__prompt__value">
                     {{ part.prompt.description|linebreaks }}
                 </div>
             </article>
             <div class="submission__answer__part__text">
-                <h2 class="submission__answer__part__text__title">{{ answer_text_label }}</h2>
+                <h3 class="submission__answer__part__text__title">{{ answer_text_label }}</h3>
```
Comment (efischer19) [inline openassessment/templates/openassessmentblock/oa_submission_answer.html]: I kept running into an overriding style coming from https://github.com/edx/edx-platform/blob/master/lms/static/sass/course/courseware/_courseware.scss#L453 for the `h2` element here.

Is changing it to an `h3` an acceptable workaround? Is there some other way to say "don't apply that style in this case" in CSS?

**Comment by efischer19** [comment]
New screenshots:

Non-leaderboard:
screen shot 2016-02-22 at 12 59 03 pm

Leaderboard:
screen shot 2016-02-22 at 12 59 25 pm

FYI @chris-mike @robrap @dianakhuang

File: openassessment/templates/openassessmentblock/oa_submission_answer.html (hunk 1 above)
Comment (cahrens) [inline openassessment/templates/openassessmentblock/oa_submission_answer.html]: Keep it as an h2, but override the rule with a more explicit rule.

File: openassessment/xblock/static/sass/oa/utilities/_shame.scss [hunk 2]
```diff
@@ -71,7 +71,16 @@
...
+  // submission answer leaderboard fix (TNL-4009). 4 class selectors is more specific than
+  // lms/static/sass/course/courseware/_courseware.scss#L453 in edx-platform (3 class selectors)
+  .leaderboard__answer .submission__answer__display__content {
+    h2.submission__answer__part__text__title {
+      @include text-align(left);
+      @include float(none);
+      font-size: inherit;
+      margin: 0;
+    }
+  }
```
Comment (efischer19) [inline openassessment/xblock/static/sass/oa/utilities/_shame.scss]: If `_shame.scss` isn't the right place for this code, let me know. This case seems to fit "rules to overcome any poorly defined, scoped, or architected edX LMS-based CSS" as described above.

File: openassessment/xblock/static/sass/oa/views/_oa-base.scss [hunk 3]
```diff
@@ -500,6 +500,10 @@
     }
   }
 
+  .submission__answer__part__text__title {
+    @extend %t-subheading;
```
Comment (efischer19) [inline openassessment/xblock/static/sass/oa/views/_oa-base.scss]: This is what's causing the bold labels - is that acceptable? I think it looks better than before.

**Comment by efischer19** [comment]
Updated to avoid `s/h2/h3` from before. Screenshots are still the same (https://github.com/edx/edx-ora2/pull/878#issuecomment-187296560), but the style definition is now made in a targeted `shame.scss` rule.

File: openassessment/xblock/static/sass/oa/utilities/_shame.scss (hunk 2 above)
Comment (cahrens) [inline openassessment/xblock/static/sass/oa/utilities/_shame.scss]: I don't see anything particularly shameful about this css. I would probably put it in _oa-base.scss, but I don't feel strongly about it.

**Comment by efischer19** [comment]
Sandbox has been rebuilt, check https://efischer19.sandbox.edx.org/courses/course-v1:edx+ORA203+course/courseware/a4dfec19cf9b4a6fb5b18be6ccd9cecc/ee5a35f57dce4b9facd7c29ca06e4597/ for a live demo.

@andy-armstrong, would you mind looking over this fix? I chose `_shame.scss` due to the existence of this (https://github.com/edx/edx-ora2/blob/master/openassessment/xblock/static/sass/oa/utilities/_shame.scss#L39), which seems similar.

**Comment by cahrens** [comment]
:+1: @dianakhuang should also re-review since the code changed quite a bit since her initial thumbs.

**Comment by dianakhuang** [comment]
:+1:

**Pull request merged**